frontend/src/metabase/common/components/FieldPicker/FieldPicker.tsx

```diff
@@ -1,28 +1,56 @@
+import { useMemo } from "react";
 import { t } from "ttag";
 import CheckBox from "metabase/core/components/CheckBox";
 import StackedCheckBox from "metabase/components/StackedCheckBox";
 import * as Lib from "metabase-lib";
 import { ToggleItem, ColumnItem } from "./FieldPicker.styled";
 
 interface FieldPickerProps {
-  columnsInfo: Lib.ColumnDisplayInfo[];
-  isAll: boolean;
-  isNone: boolean;
-  isDisabledDeselection?: boolean;
+  query: Lib.Query;
+  stageIndex: number;
+  columns: Lib.ColumnMetadata[];
+  "data-testid"?: string;
+  isColumnSelected?: (column: Lib.ColumnMetadata) => boolean;
   onToggle: (columnIndex: number, isSelected: boolean) => void;
   onSelectAll: () => void;
   onSelectNone: () => void;
 }
 
 export const FieldPicker = ({
-  columnsInfo,
-  isAll,
-  isNone,
-  isDisabledDeselection,
+  query,
+  stageIndex,
+  columns,
   onToggle,
   onSelectAll,
   onSelectNone,
+  isColumnSelected = (column: Lib.ColumnMetadata) =>
```

Comment:
Could we pass this in the fields step instead?

Reply:
@ranquild `isColumnSelected` is an optional prop and it's a default implementation here, i.e. it's still possible to pass a different function on a need (like it's done in `JoinTablePicker`). Or do you have something else in mind here?

Reply:
The component should be used in 2 places now - in fields and joins. I thought that we could require this function and pass the implementation in the fields clause